### ISSUE TITLE
Improve AADL property editing support in graphical editor

### ISSUE THREAD

**Issue by philip-alldredge** [osate/osate2#1169]
<a href="https://github.com/philip-alldredge"><img src="https://avatars2.githubusercontent.com/u/1376364?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [philip-alldredge](https://github.com/philip-alldredge)**
_Thursday Oct 15, 2015 at 15:54 GMT_
_Originally opened as https://github.com/osate/osate-ge/issues/140_

----

> allow the user to assign AADL property values.

Problems assigning properties to flow paths using the graphical editor. Revisit integration with properties view. Ideally support property associations with applies to along any depth of the model.

A way is needed to visualize and edit properties.

Additionally, specialty menus are needed to set common properties. For example: Delayed/Sampled/Immediate connections. Need to allow user to set it at the desired level without being tedious. Other properties should be supported as well. The AADL properties view has grouping of properties.

Look into how the graphical editor support for AADL properties can be improved and be consistent and potentially build on the capabilities provided by the OSATE core.

**Comment by philip-alldredge** [comment]
<a href="https://github.com/philip-alldredge"><img src="https://avatars2.githubusercontent.com/u/1376364?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [philip-alldredge](https://github.com/philip-alldredge)**
_Wednesday May 04, 2016 at 16:13 GMT_

----

There is also an additional issue that if the xtext document is not open, the AADL Properties View does not work. Currently the graphical editor does not provide an XtextDocument but rather edits the resource directly unless the AADL editor is open. Look into whether having an XtextDocument is an option. Could also reduce editing code paths.

**Comment by philip-alldredge** [comment]
<a href="https://github.com/philip-alldredge"><img src="https://avatars2.githubusercontent.com/u/1376364?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [philip-alldredge](https://github.com/philip-alldredge)**
_Wednesday Feb 28, 2018 at 15:50 GMT_

----

Need a general mechanism to set properties on end-to-end flows and flow implementations. It has also been suggested that it would be nice to have an easy way to set the latency property when creating such flows.

Is there some way to unify how ETEF and Flow Implementations are handled so that they would be exposed to the properties view and/or outline?

**Comment by philip-alldredge** [comment]
Consider adding multiple selection.

**Comment by philip-alldredge** [comment]
If support for the text editor's OSATE's AADL Property Values view is not fixed as part of the task, the integration should be completely disabled.

**Comment by philip-alldredge** [comment]
@lwrage @joeseibel:
@RyanMcilnay and I have discussed things and we are looking at focusing on updating things so that the OSATE AADL Property Values view works with the graphical editor in the same was it does for the text editor. Multiple-selection and handling the context information of the graphical editor is something we can look at once the basic functionality is complete.
